### PR TITLE
Add Gradle support and trigger docker build if Dockerfile is present

### PR DESCRIPTION
**Pull request by lievendoclo**

### CODE REVIEW

File: lib/stack/GradleBuildInterpreter.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        });
+
+    private readonly gradleVersionGoal: Version = new Version()
+        .with({
+            ...GradleDefaultOptions,
+            versioner: GradleProjectVersioner,
+        });
+
+    private readonly dockerBuildGoal: DockerBuild = new DockerBuild()
+        .with({
```
Comment (cdupuis) [inline lib/stack/GradleBuildInterpreter.ts:72]: progress reporter and dockerfile finder missing

Comment (lievendoclo) [reply lib/stack/GradleBuildInterpreter.ts:72]: Fixed

File: lib/stack/buildSystemScanner.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * Stack information on the used build system. Currently supports Gradle and Maven.
+ */
+export interface BuildSystemStack extends TechnologyStack {
+
+    name: "javabuild";
+
+    /**
+     * Which build system is used.
+     */
+    buildSystem: "maven" | "gradle";
```
Comment (cdupuis) [inline lib/stack/buildSystemScanner.ts]: Can this be an enum please?

File: lib/stack/buildSystemScanner.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        dockerFile = f.path;
+    });
+
+    if (!isMaven && !isGradle) {
+        return undefined;
+    }
+
+    const stack: BuildSystemStack = {
+        name: "javabuild",
+        buildSystem: isGradle ? "gradle" : "maven",
```
Comment (cdupuis) [inline lib/stack/buildSystemScanner.ts]: With an enum this is easier to type.

**Review by cdupuis — APPROVED** [review]

**Pull request merged**